cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/functions/AnyColumn.java

```diff
@@ -0,0 +1,27 @@
+package com.ibm.cohort.cql.functions;
+
+import java.util.stream.Collectors;
+
+import com.ibm.cohort.datarow.model.DataRow;
+
+public class AnyColumn {
+
+    private AnyColumn() {
+    }
+
+    public static Object AnyColumn(Object object, String fieldPrefix) {
+        DataRow dataRow = (DataRow) object;
+
+        return dataRow.getFieldNames().stream().filter(fieldName -> fieldName.startsWith(fieldPrefix))
+            .map(dataRow::getValue)
+            .collect(Collectors.toList());
+    }
+
+    public static Object AnyColumnRegex(Object object, String regex) {
+        DataRow dataRow = (DataRow) object;
+
+        return dataRow.getFieldNames().stream().filter(fieldName -> fieldName.matches(regex))
```

Comment:
`String.matches` will recompile the regex every time it is called, so it makes me a little nervous using it in a place that will get called millions of times. However, given that the context is matching field names and I expect those patterns to be relatively trivial, I expect the difference is fairly negligible. We might consider implementing a regex cache here if it becomes a problem.